test/gtest/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+include(googletest) #include googletest.cmake
+enable_testing()
+include(GoogleTest)
+
+#add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND} --output-on-failure -C ${CMAKE_CFG_INTDIR})
+#add_custom_target(tests)
+
+function(add_gtest TEST_NAME)
+  message("Adding Test: " ${TEST_NAME})
+  add_executable(test_${TEST_NAME} ${TEST_NAME}.cpp )
+  add_dependencies(tests test_${TEST_NAME})
+  add_dependencies(check test_${TEST_NAME})
+  target_compile_options(test_${TEST_NAME} PRIVATE -Wno-global-constructors -Wno-undef)
```

Comment:
Adding the include directory would remove the requirement of up-leveling each directory
```
target_include_directories(test_${TEST_NAME} PRIVATE ../)
```

Reply:
Thanks and done